gwcs/wcs.py

```diff
@@ -519,31 +519,65 @@ def __repr__(self):
             self.output_frame, self.input_frame, self.forward_transform)
         return fmt
 
-    def footprint(self, bounding_box=None, center=False):
+    def footprint(self, bounding_box=None, center=False, kind="all"):
         """
-        Return the footprint of the observation in world coordinates.
+        Return the footprint in world coordinates.
 
         Parameters
         ----------
         bounding_box : tuple of floats: (start, stop)
             `prop: bounding_box`
         center : bool
             If `True` use the center of the pixel, otherwise use the corner.
+        kind : str
+            A supported ``output_frame.axes_type`` or "all" (default).
+            One of ['spatial', 'spectral', 'temporal'] or a custom type.
 
         Returns
         -------
-        coord : array of coordinates in the output_frame.
-            The order is counter-clockwise starting with the bottom left corner.
+        coord : ndarray
+            Array of coordinates in the output_frame mapping
+            corners to the output frame. For spatial coordinates the order
+            is clockwise.
```

Comment:
Does the output here also start from the bottom left corner?  Might be good to make this clear.